### PR TITLE
Transfered the trajectory passing in autonsystem to be handled by ros  messages

### PR DESCRIPTION
**Pull request by mehulgoel873**
## What type of PR is this? (check all applicable)

- [X] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Changes -- I created a pack and unpack function within the trajectory class that takes the positions and packs it into a ros msg that can be sent over  ros topics. 

The reasoning behind this is that this better separates the controller from the path planner to better reflect our software architecture and will make more sense overall.

The benefits behind this change is that the overall repository now has a chance for becoming cleaner and overall more likely to work.

## Related Tickets & Documents

- Related Issue #
- Closes #

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._

## Added/updated tests?
_We encourage you to keep the code coverage percentage at 80% and above._

I ran the single and double version of the buggy sim, and the code seemed to be working.

- [X ] Yes
- [] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: rb_ws/src/buggy/scripts/auton/autonsystem.py [hunk 1]
```diff
@@ -252,7 +261,7 @@ def planner_tick(self):
             other_pose = self.get_world_pose(self.other_odom_msg)
 
         # update local trajectory via path planner
-        self.cur_traj, cur_idx = self.path_planner.compute_traj(
+        _, cur_idx = self.path_planner.compute_traj(
```
Comment (TiaSinghania) [inline rb_ws/src/buggy/scripts/auton/autonsystem.py:264]: if the goal was to decouple pathplanning and autonsystem, why are we returning cur_idx

**Review by TiaSinghania — COMMENTED** [review]
Is this the final version of the trajectory setup? It seems like the files are still as coupled as before, how does prompting the trajectory class to publish a new trajectory each tick different from just having it return the trajectory directly?

File: rb_ws/src/buggy/scripts/auton/autonsystem.py (hunk 1 above)
Comment (mehulgoel873) [reply rb_ws/src/buggy/scripts/auton/autonsystem.py:264]: You are correct, I am about to push a commit that fixes this bug. I must've missed it in my first pass through.

**Comment by mehulgoel873** [comment]
> Is this the final version of the trajectory setup? It seems like the files are still as coupled as before, how does prompting the trajectory class to publish a new trajectory each tick different from just having it return the trajectory directly?

This is correct, I haven't decoupled the files in the autonsystem just yet. This requires recreating how we treat the controllers and path planners, which I wanted to make a separate task due to the behemoth that is. However, since now all the information passed between both the controller and path planner happens only through a ros topic, it should be relatively simple to separate both into different ros nodes.

**Review by TiaSinghania — APPROVED** [review]

**Pull request merged**